### PR TITLE
Spring cleaning

### PR DESCRIPTION
**Pull request by nmay231**
* **Please check if the PR fulfills these requirements**
- [x] The commit message(s) are descriptive of the changes made
- [x] The PR contains changes that are focused and differs from other PRs

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

- Removed code smells/wrote code to be clearer
- Added sections to issue and PR templates for doc changes

### CODE REVIEW

**Comment by nmay231** [comment]
@omnidan These are just a few minor changes.

I think I'm going to work on field extensions next (you can look at a demo in [field-extensions-demo](https://github.com/omnidan/redux-undo/compare/field-extensions-demo))

File: src/reducer.js [hunk 1]
```diff
@@ -14,22 +14,17 @@ function createHistory (state, ignoreInitialState) {
...
-  return history.past.length + 1
-}
-
 // insert: insert `state` into history, which means adding the current state
 //         into `past`, setting the new `state` as `present` and erasing
 //         the `future`.
 function insert (history, state, limit, group) {
   debug.log('inserting', state)
-  debug.log('new free: ', limit - lengthWithoutFuture(history))
+  debug.log('new free: ', limit - history.past.length - 1)
```
Comment (omnidan) [inline src/reducer.js]: I think now it is a bit unclear why we are subtracting one. Either we store this in a separate variable (`history.path.length + 1`), or we keep the function

**Comment by nmay231** [comment]
Changes made :+1:

I already rebased it manually, so this can be merged normally. Thanks.

**Comment by omnidan** [comment]
Thanks @nmay231 !

**Pull request merged**